server/index.js

```diff
@@ -51,6 +62,28 @@ app.get('/api/trip/:tripId', (req, res, next) => {
     });
 });
 
+app.get('/api/itinerary/:tripId', (req, res, next) => {
+  const tripId = parseInt(req.params.tripId, 10);
+  if (!tripId) {
+    throw new ClientError(400, 'tripId must be a positive integer');
+  }
+  const sql = `
+  select *
+  from "itinerary"
+  where "tripId" = $1
+  order by "timeStart"`;
+
+  const params = [tripId];
+
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        throw new ClientError(404, `cannot find trip with tripId ${tripId}`);
+      }
+      res.json(result.rows);
```

Comment:
If possible, avoid sending the client an array when they only need one object.

Reply:
It looks like I'm just pulling one object, but I'm actually pulling multiple objects from the itinerary db. There can be multiple objects to the tripId.

Reply:
Ok, i see!